File: code/0849/849.到最近的人的最大距离.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int res = 0, n = seats.size();

        for (int i = 0; i < n; i++) {
            if (seats[i]) continue;
            int j = i + 1;
            while (j < n && !seats[j]) j++;
            if (!i || j == n)
                res = max(res, j - i);
            else res = max(res, (j - i + 1) / 2);
        }

        return res;
    }
};

File: code/0849/849.到最近的人的最大距离.py
class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        res, l, n = 0, 0, len(seats)

        while l < n and seats[l] == 0:
            l += 1
        res = max(res, l)

        while l < n:
            r = l + 1
            while r < n and seats[r] == 0:
                r += 1
            
            if r == len(seats):
                res = max(res, r - l - 1)
            else:
                res = max(res, (r - l) // 2)
            
            l = r
        
        return res
